iml-mailbox/src/lib.rs

```diff
@@ -29,148 +43,77 @@ fn streamer<'a>(
-///
-/// It will then write any incoming lines from the passed `mpsc::UnboundedReceiver`
-/// to that file.
-pub async fn ingest_data(
-    address: PathBuf,
-    mut rx: mpsc::UnboundedReceiver<Incoming>,
-) -> Result<(), std::io::Error> {
-    tracing::info!("Starting ingest for {:?}", address);
-
-    let mut file = OpenOptions::new()
-        .append(true)
-        .create(true)
-        .open(address)
-        .await?;
-
-    while let Some(incoming) = rx.next().await {
-        match incoming {
-            Incoming::Line(mut line) => {
-                if !line.ends_with('\n') {
-                    line.extend(['\n'].iter());
-                }
-
-                tracing::trace!("handling line {:?}", line);
+    let pool = iml_orm::pool()?;
 
-                file.write_all(line.as_bytes()).await?;
-            }
-            Incoming::EOF(tx) => {
-                file.flush().await?;
+    let task = match get_task_by_name(&task, &pool).await? {
+        Some(t) => t,
+        None => {
+            tracing::error!("Task {} not found", &task);
 
-                let _ = tx.send(());
-            }
+            return Err(MailboxError::NotFound(format!("Failed to find {}", &task)));
         }
-    }
-
-    file.flush().await?;
-
-    Ok(())
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use std::fs;
-    use tempdir::TempDir;
-
-    #[tokio::test]
-    async fn test_mailbox_senders() -> Result<(), Box<dyn std::error::Error>> {
-        let tmp_dir = TempDir::new("test_mailbox")?;
-        let address = tmp_dir.path().join("test_message_1");
-
-        let mut mailbox_sender = MailboxSenders::default();
-
-        let (tx, fut) = mailbox_sender.create(address.clone());
-
-        tx.unbounded_send(Incoming::Line("foo\n".into()))?;
-
-        mailbox_sender
-            .get(&address)
-            .unwrap()
-            .unbounded_send(Incoming::Line("bar".into()))?;
-
-        tx.unbounded_send(Incoming::Line("baz\n".into()))?;
-
-        mailbox_sender.remove(&address);
-
-        drop(tx);
+    };
+
+    let xs = lines.iter().filter_map(|line| {
+        tracing::trace!("handling line {:?}", &line);
+
+        let mut map: HashMap<String, serde_json::Value> = serde_json::from_str(&line).ok()?;
+
+        if let Some(fid) = map.remove("fid".into()) {
+            let fid = LustreFid::from_str(
+                fid.as_str()
+                    .ok_or_else(|| {
+                        MailboxError::NotFound(format!("Failed to find fid in {}", line))
+                    })
+                    .ok()?,
+            )
+            .ok()?;
+
+            // Get "other" data if it exists
+            let data = match map.iter().next() {
+                Some((_, v)) => v.clone(),
+                None => json!({}),
+            };
+
+            tracing::trace!("Inserting fid:{:?} data:{:?} task:{:?}", fid, data, task);
+
+            Some(insert_fidtask(fid, data, &task).execute_async(&pool))
```

Comment:
I wonder how much of a bottleneck doing one write at a time is instead of doing a bulk write.